service/history/events/cache.go

```diff
@@ -210,3 +218,21 @@ func (e *CacheImpl) getHistoryEventFromStore(
 
 	return nil, errEventNotFoundInBatch
 }
+
+func (e *CacheImpl) put(key EventKey, event *historypb.HistoryEvent) interface{} {
+	return e.Put(key, NewHistoryEventCacheItem(event))
+}
+
+var _ historyEventItem = (*historyEventItemImpl)(nil)
+
+func NewHistoryEventCacheItem(
```

Comment:
this constructor should not be exported I think. it's the impl detail of history events cache